packages/my-queries-storage/src/hooks.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect } from 'react';
+import { useSyncExternalStore } from 'use-sync-external-store/shim';
+import type { FavoriteQuerySchema, RecentQuerySchema } from './query-storage';
```

Comment:
You're not supposed to be using zod schemas like this. This should be using exported schema output types, not the schemas directly

Reply:
Ah, didn't know, thanks! I was just drafting, TBH, don't consider any of this code production in any of the dimensions it could be production ready 😄